### PR TITLE
Replace `in-addr.arpa` records in `zone` label for `coredns_cache_.*`

### PR DESCRIPTION
**Pull request by kopiczko**
See the slack thread for more details https://gigantic.slack.com/archives/C02GLN41UP2/p1719305781626429

## Regexp test

```go
package main

import (
	"fmt"
	"regexp"
)

const expr = `([^,]*?,?)(?:(?:,?[\d\.]+\.in-addr\.arpa\.)+)+(,?.*)`

var tests = []string{
	`cluster.local.,31.172.in-addr.arpa.,64.100.in-addr.arpa.,65.100.in-addr.arpa.,something.else.`,
	`cluster.local.,31.172.in-addr.arpa.,64.100.in-addr.arpa.,65.100.in-addr.arpa.,66.100.in-addr.arpa.`,
	`cluster.local.,something.else.`,
	`31.172.in-addr.arpa.,64.100.in-addr.arpa.,65.100.in-addr.arpa.,66.100.in-addr.arpa.,something.else.`,
	`31.172.in-addr.arpa.,64.100.in-addr.arpa.,65.100.in-addr.arpa.,66.100.in-addr.arpa.`,
}

func main() {
	re := regexp.MustCompile("^" + expr + "$")

	for _, test := range tests {
		matches := re.FindAllStringSubmatch(test, -1)
		if len(matches) > 1 {
			panic(fmt.Sprintf("found %d matches, wanted max 1", len(matches)))
		}

		fmt.Printf("\nGroups for %q:\n", test)
		if len(matches) == 0 {
			fmt.Printf("\t__ does not match __\n")
			continue
		}
		for i, group := range matches[0][1:] {
			fmt.Printf("\t%d: %s\n", i+1, group)
		}
	}
}
```

```
$ go run test_regexp.go

Groups for "cluster.local.,31.172.in-addr.arpa.,64.100.in-addr.arpa.,65.100.in-addr.arpa.,something.else.":
        1: cluster.local.,
        2: ,something.else.

Groups for "cluster.local.,31.172.in-addr.arpa.,64.100.in-addr.arpa.,65.100.in-addr.arpa.,66.100.in-addr.arpa.":
        1: cluster.local.,
        2: 

Groups for "cluster.local.,something.else.":
        __ does not match __

Groups for "31.172.in-addr.arpa.,64.100.in-addr.arpa.,65.100.in-addr.arpa.,66.100.in-addr.arpa.,something.else.":
        1: 
        2: ,something.else.

Groups for "31.172.in-addr.arpa.,64.100.in-addr.arpa.,65.100.in-addr.arpa.,66.100.in-addr.arpa.":
        1: 
        2: 
```

## On `grizzly`

Screenshot 2024-06-26 at 14 06 30

### CODE REVIEW

**Review by QuentinBisson — APPROVED** [review]

**Review by QuentinBisson — CHANGES_REQUESTED** [review]
We should use labeldrop instead

File: helm/kube-prometheus-stack/values.yaml [hunk 1]
```diff
@@ -45,6 +45,13 @@ kube-prometheus-stack:
         regex: coredns_dns_(request_size_bytes_bucket|response_size_bytes_bucket)
         sourceLabels:
         - __name__
+      # Drop zone label, due to large cardinality. Zones can be deducted from
+      # the server configuration and we keep the server label.
+      - action: LabelDrop
```
Comment (QuentinBisson) [inline helm/kube-prometheus-stack/values.yaml:50]:
```suggestion
      - action: labeldrop
```

**Comment by kopiczko** [comment]
Tested on `grizzly`:

Screenshot 2024-06-26 at 14 06 30

**Review by QuentinBisson — APPROVED** [review]
LGTM

**Review by mcharriere — APPROVED** [review]
lgtm!

**Pull request merged**